sw/host/opentitanlib/src/image/image.rs

```diff
@@ -97,6 +93,28 @@ impl Image {
         Ok(())
     }
 
+    /// Adds a new extension at the end of the image and updates size.
+    /// Note that this method does NOT update the `signed_region_end` or `length` fields of the
+    /// manifest.
+    /// New extensions are ALWAYS added at the end of the image by design.
+    pub fn add_extension_data(&mut self, bytes: &[u8]) -> Result<()> {
+        self.data.bytes[self.size..self.size + bytes.len()].copy_from_slice(bytes);
```

Comment:
These functions return `Result`s already, we probably shouldn't panic on out-of-range slice accesses. You can use `get_mut()` along with `ok_or()` on a new `ImageError` variant.

Reply:
good point, done, thanks!